tests/installation/snapper_rollback_test.pm

```diff
@@ -0,0 +1,67 @@
+# SUSE's openQA tests
+#
+# Copyright © 2009-2013 Bernhard M. Wiedemann
+# Copyright © 2012-2016 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+
+### the test's goal is to test a rollback (as a backup) after system has migrated.
+### we make one rollback to going back to downgraded system, (1)
+### and we make another rollback to go to upgraded system. at the end 6 snapshots.
+### we are in a read_only_snapsot 1 and after we are in a read_and_write 2
+
+### TOFIX: we could make a cleanup test, to remove the 4 created snapshots.
+use base "consoletest";
+use testapi;
+use utils;
+
+
+sub run() {
+    my $self = shift;
+    #1) read only snapshot! don't try to save something here.
+    script_run("snapper rollback -d rollback-before-migration");
+
+    assert_script_run("snapper list | tail -n 2 | grep rollback");
+
+    # we are in the before migration version. so snap 4 is identic as snap 1
+    ## 1 before migration = 4. since we don't have already booted on snapshot.
+    # after normal_booting in 4, there are login stuff that differ from 1.
+    script_run("echo \"SNAPSHOT_1&4_IDENTIC\" >/dev/$serialdev");
+    script_run("snapper diff 1..4 >/dev/$serialdev");
+    script_run("snapper status 1..4>/dev/$serialdev");
+    wait_serial('SNAPSHOT_1&4_IDENTIC') || die "CRITICAL::SNAPSHOTS DIFFER! MUST BE THE SAME";
```

Comment:
I don't really understand the logic here, but before explaining note that this approach will no longer work. `script_run` without timeout steals serial output now.
When you use `script_run` and then `wait_serial` add 0 timeout e.g.: 
```perl
script_run("snapper status 1..4 >/dev/$serialdev", 0);
wait_serial($something) || die "$something not found";
```